src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_gligen.py

```diff
@@ -0,0 +1,827 @@
+            boxes = boxes.unsqueeze(0).expand(repeat_batch, -1, -1).clone()
+            text_embeddings = text_embeddings.unsqueeze(0).expand(repeat_batch, -1, -1).clone()
+            masks = masks.unsqueeze(0).expand(repeat_batch, -1).clone()
+            if do_classifier_free_guidance:
+                masks[: repeat_batch // 2] = 0
+            if cross_attention_kwargs is None:
+                cross_attention_kwargs = {}
+            cross_attention_kwargs["gligen"] = {"boxes": boxes, "positive_embeddings": text_embeddings, "masks": masks}
+
+            # Prepare latent variables for GLIGEN inpainting
+            if gligen_inpaint_image is not None:
+                # if the given input image is not of the same size as expected by VAE
+                # center crop and resize the input image to expected shape
+                if gligen_inpaint_image.size != (self.vae.sample_size, self.vae.sample_size):
+                    gligen_inpaint_image = self.target_size_center_crop(gligen_inpaint_image, self.vae.sample_size)
+
+                gligen_inpaint_image = torch.from_numpy(np.asarray(gligen_inpaint_image))
+                # Convert a single image into a batch of images with a batch size of 1
+                # The resulting shape becomes (1, C, H, W), where C is the number of channels,
+                # and H and W are the height and width of the image.
+                gligen_inpaint_image = gligen_inpaint_image.unsqueeze(0).permute(0, 3, 1, 2)
+                # scales the pixel values to a range [-1, 1]
+                gligen_inpaint_image = gligen_inpaint_image.to(dtype=torch.float32) / 127.5 - 1.0
+                gligen_inpaint_image = gligen_inpaint_image.to(dtype=self.vae.dtype, device=self.vae.device)
+                # Run AutoEncoder to get corresponding latents
+                gligen_inpaint_latent = self.vae.encode(gligen_inpaint_image).latent_dist.sample()
+                gligen_inpaint_latent = self.vae.config.scaling_factor * gligen_inpaint_latent
+                # Generate an inpainting mask
+                # pixel value = 0, where the object is present (defined by bounding boxes above)
+                #               1, everywhere else
+                gligen_inpaint_mask = self.draw_inpaint_mask_from_boxes(
+                    _boxes[:n_objs], gligen_inpaint_latent.shape[2:]
+                )
+                gligen_inpaint_mask = gligen_inpaint_mask.to(
+                    dtype=gligen_inpaint_latent.dtype, device=gligen_inpaint_latent.device
+                )
+                gligen_inpaint_mask = gligen_inpaint_mask[None, None]
+                gligen_inpaint_mask_addition = torch.cat(
+                    (gligen_inpaint_latent * gligen_inpaint_mask, gligen_inpaint_mask), dim=1
+                )
+                # Convert a single mask into a batch of masks with a batch size of 1
+                gligen_inpaint_mask_addition = gligen_inpaint_mask_addition.expand(repeat_batch, -1, -1, -1).clone()
```

Comment:
As I mentioned above, 
using `batch_size * num_images_per_prompt` wouldn't be right in `do_classifier_free_guidance` case. We don't need this change because of that